Disjoint SmartGraphs (#429)
arangodb/docs: 3.7/graphs-smart-graphs.md

However the overall performance is never worse than the same query using a
General Graph.

Benefits of DisjointSmartGraphs
-----------------------

> DisjointSmartGraphs are a specialized type of SmartGraphs. 

Reviewer: ```suggestion
Disjoint SmartGraphs are a specialized type of SmartGraphs. 
```

> In addition to the guaranteed sharding in SmartGraphs, a DisjointSmartGraph

Reviewer: ```suggestion
In addition to the guaranteed sharding in SmartGraphs, a Disjoint SmartGraph
```

> prohibits edges between vertices with different smartGraphAttribute values.

Reviewer: ```suggestion
prohibits edges between vertices with different `smartGraphAttribute` values.
```

> This ensures graph traversals, shortest path, and k-shortest-paths queries

Reviewer: ```suggestion
This ensures that graph traversals, shortest path, and k-shortest-paths queries
```

can be executed locally on a DB-Server, achieving improved performance for

> this type of query.

Reviewer: ```suggestion
these type of queries.
```

However the overall performance is never worse than the same query using a
General Graph.

> Benefits of DisjointSmartGraphs

Reviewer 2: What is the spelling? I assumed DisjointSmartGraphs

Author: DisjointSmartGraphs is correct.

-----------------------

> DisjointSmartGraphs are a more specialized type of SmartGraphs. Compared to 

Reviewer 3: ```suggestion
DisjointSmartGraphs are a specialized type of SmartGraphs. 
```

> SmartGraphs they are quite equal compared to their origin, expect one important
> detail:

Reviewer 3: ```suggestion
```

Where SmartGraphs are almost local, DisjointSmartGraphs are guaranteed to be

> local in any case by definition. A DisjointSmartGraph prohibids connected edges

Reviewer 2: prohibits

different SmartGraph components. As no connections between subgraphs are
existing, no network overhead will ever appear, which will lead to faster
query traversal execution times. Besides that difference, the behaviour of
all other functionalities are the same.

> 

Reviewer 3: In addition to the guaranteed sharding in SmartGraphs, a DisjointSmartGraph prohibits edges between vertices with different smartGraphAttribute values.

This ensures graph traversals, shortest path, and k-shortest-paths queries can be executed locally on a DBServer, achieving improved performance for this type of query.

{% endarangoshexample %}
{% include arangoshexample.html id=examplevar script=script result=result %}

> **Create a disjoint graph**

Reviewer: ```suggestion
**Create a Disjoint SmartGraph**
```

In contrast to regular SmartGraphs we have to add one option when creating the
graph. The boolean option `isDisjoint` is required, needs to be set to `true`
and cannot be modified later. 

{% arangoshexample examplevar="examplevar" script="script" result="result" %}

>     @startDocuBlockInline smartGraphCreateGraphHowTo1_disjoint_cluster
>     @EXAMPLE_ARANGOSH_OUTPUT{smartGraphCreateGraphHowTo1_disjoint_cluster}

Reviewer: ```suggestion
    @startDocuBlockInline smartGraphCreateGraphHowToDisjoint1_cluster
    @EXAMPLE_ARANGOSH_OUTPUT{smartGraphCreateGraphHowToDisjoint1_cluster}
```

      var graph_module = require("@arangodb/smart-graph");
      var graph = graph_module._create("myGraph", [], [], {smartGraphAttribute: "region", numberOfShards: 9, isDisjoint: true});
      graph_module._graph("myGraph");
     ~graph_module._drop("myGraph");
    @END_EXAMPLE_ARANGOSH_OUTPUT

>     @endDocuBlock smartGraphCreateGraphHowTo1_disjoint_cluster

Reviewer: ```suggestion
    @endDocuBlock smartGraphCreateGraphHowToDisjoint1_cluster
```
